compute-nodes.yaml

```
@@ -0,0 +1,94 @@
+- import_playbook: common.yaml
+
+- hosts: all
+  gather_facts: no
+
+  tasks:
+  # Build a list of worker nodes with indexes
+  - name: Build worker list
+    set_fact:       
+      worker_list: "{{ worker_list | default([]) + [ item.1 + '-' + item.0 | string ] }}"
+    with_indexed_items: "{{ [os_port_worker] * os_compute_nodes_number }}"
+
+  # Create the SDN/primary port for each worker node
+  - name: 'Create the Compute ports'
+    os_port:
+      name: "{{ item }}"
+      network: "{{ os_network }}"
+      security_groups:
+      - "{{ os_sg_worker }}"
+      allowed_address_pairs:
+      - ip_address: "{{ os_ingressVIP }}"
+    with_items: "{{ worker_list }}"
+    register: ports
```

`ports` is a bit too vague, I would name it `compute_ports` in case we have other types of ports later.

```suggestion
    register: compute_ports
```

---

That's what we do in the UPI playbook, so actually I would leave this for now :)